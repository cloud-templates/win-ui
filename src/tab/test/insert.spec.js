import { mount, later } from '../../../test';

// this case will throw wierd error in index.spec.js
// so separate it
test('insert tab dynamically', async () => {
  const wrapper = mount({
    template: `
      <win-tabs v-model="active">
        <win-tab title="1">1</win-tab>
        <div v-if="insert">
          <win-tab title="2">2</win-tab>
        </div>
        <win-tab title="3">3</win-tab>
      </win-tabs>
    `,
    data() {
      return {
        insert: false,
        active: 1,
      };
    },
  });

  await later();
  wrapper.setData({ insert: true });
  expect(wrapper).toMatchSnapshot();
});

// this case will throw wierd error in index.spec.js
// so separate it
test('insert tab with child component', async () => {
  const wrapper = mount({
    template: `
      <win-tabs v-model="active">
        <win-tab title="1">1</win-tab>
        <my-tab />
        <win-tab title="3">3</win-tab>
      </win-tabs>
    `,
    components: {
      'my-tab': {
        template: `<win-tab title="2">2</win-tab>`,
      },
    },
  });

  await later();
  expect(wrapper).toMatchSnapshot();
});
