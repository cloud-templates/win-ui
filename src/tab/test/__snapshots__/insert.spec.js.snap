// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert tab dynamically 1`] = `
<div class="win-tabs win-tabs--line">
  <div class="win-tabs__wrap win-hairline--top-bottom">
    <div role="tablist" class="win-tabs__nav win-tabs__nav--line">
      <div role="tab" class="win-tab"><span class="win-tab__text win-tab__text--ellipsis">1</span></div>
      <div role="tab" aria-selected="true" class="win-tab win-tab--active"><span class="win-tab__text win-tab__text--ellipsis">2</span></div>
      <div role="tab" class="win-tab"><span class="win-tab__text win-tab__text--ellipsis">3</span></div>
      <div class="win-tabs__line" style="width: 0px; transform: translateX(0px) translateX(-50%);"></div>
    </div>
  </div>
  <div class="win-tabs__content">
    <div role="tabpanel" class="win-tab__pane" style="display: none;">
      <!---->
    </div>
    <div>
      <div role="tabpanel" class="win-tab__pane">2</div>
    </div>
    <div role="tabpanel" class="win-tab__pane" style="display: none;">3</div>
  </div>
</div>
`;

exports[`insert tab with child component 1`] = `
<div class="win-tabs win-tabs--line">
  <div class="win-tabs__wrap win-hairline--top-bottom">
    <div role="tablist" class="win-tabs__nav win-tabs__nav--line">
      <div role="tab" aria-selected="true" class="win-tab win-tab--active"><span class="win-tab__text win-tab__text--ellipsis">1</span></div>
      <div role="tab" class="win-tab"><span class="win-tab__text win-tab__text--ellipsis">2</span></div>
      <div role="tab" class="win-tab"><span class="win-tab__text win-tab__text--ellipsis">3</span></div>
      <div class="win-tabs__line"></div>
    </div>
  </div>
  <div class="win-tabs__content">
    <div role="tabpanel" class="win-tab__pane">1</div>
    <div role="tabpanel" class="win-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="win-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;
